src/codeflare_sdk/ray/cluster/cluster.py

```diff
@@ -176,6 +186,50 @@ def up(self):
         except Exception as e:  # pragma: no cover
             return _kube_api_error_handling(e)
 
+    def apply(self, force=False):
+        """
+        Applies the Cluster yaml using server-side apply.
+        If 'force' is set to True, conflicts will be forced.
+        """
+        # check if RayCluster CustomResourceDefinition exists if not throw RuntimeError
+        self._throw_for_no_raycluster()
+        namespace = self.config.namespace
+        # Ensure Kubernetes configuration is loaded
+        try:
+            self.config_check()
+            # Get the RayCluster custom resource definition
+            api = self.get_dynamic_client().resources.get(
+                api_version="ray.io/v1", kind="RayCluster"
+            )
+        except Exception as e:
+            raise RuntimeError("Failed to get RayCluster resource: " + str(e))
+
+        # Read the YAML file and parse it into a dictionary
+        try:
+            with open(self.resource_yaml, "r") as f:
+                resource_body = yaml.safe_load(f)
+        except FileNotFoundError:
+            raise RuntimeError(f"Resource YAML file '{self.resource_yaml}' not found.")
+        except yaml.YAMLError as e:
+            raise ValueError(f"Failed to parse resource YAML: {e}")
```

Comment:
This `apply()` function only seems to work with written resources.
Have you thought about adding in support for in-memory resources like in [up()](https://github.com/project-codeflare/codeflare-sdk/blob/main/src/codeflare_sdk/ray/cluster/cluster.py#L503-L508)